File: WebRTC-iOS-ojc/WebRTC-iOS-ojc/ViewController.m
//
//  ViewController.m
//  WebRTC-iOS-ojc
//
//  Created by Innovation on 4/12/17.
//  Copyright © 2017 Innovation. All rights reserved.
//

#import "ViewController.h"

#define SERVER_URL "wss://iconthin.com:8443/ws/"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UIView *localVideoContainer;
@property (weak, nonatomic) IBOutlet UIView *remoteVideoContainer;

@property (weak, nonatomic) IBOutlet UITextField *roomIdField;

@property (nonatomic) BOOL audioOn;
@property (nonatomic) BOOL videoOn;
@property (nonatomic) BOOL useSpeaker;

@end

@implementation ViewController

- (void)viewDidLoad {
    
    [RTCClientConfig loadWithConfig:@{@"audioOnly": @YES}];
    
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    _audioOn = YES;
    _videoOn = YES;
    _useSpeaker = YES;
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)connectClick:(id)sender {
    [_roomIdField resignFirstResponder];
    
    if(_roomIdField.text.length > 0){
        NSString* roomId = [_roomIdField.text stringByReplacingOccurrencesOfString:@" " withString:@""];
        [RTCClient.shared connectWithServerUrl:@SERVER_URL roomId:roomId delegate:self];
    }else{
        // Show an alert, useless code
        UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Sorry" message:@"Please enter an roomId"preferredStyle:UIAlertControllerStyleAlert];
        UIAlertAction *actionOk = [UIAlertAction actionWithTitle:@"Ok" style:UIAlertActionStyleDefault handler:nil];
        [alertController addAction:actionOk];
        [self presentViewController:alertController animated:YES completion:nil];
    }
}
- (IBAction)audioClick:(id)sender {
    _audioOn = !_audioOn;
    [RTCClient.shared setAudioOn:_audioOn];
}
- (IBAction)videoClick:(id)sender {
    _videoOn = !_videoOn;
    [RTCClient.shared setVideoOn:_videoOn];

}
- (IBAction)speakerClick:(id)sender {
    _useSpeaker = !_useSpeaker;
    [RTCClient.shared setAudioOutputWithUseSpeaker:_useSpeaker];
}
- (IBAction)disconnectClick:(id)sender {
    [RTCClient.shared disconect];
}

#pragma mark - RTCClientDelegate metods impl

- (void)rtcClientDidSetLocalMediaStreamWithClient:(RTCClient *)client authorized:(BOOL)authorized audioOnly:(BOOL)audioOnly{
    if(authorized){
        [client setLocalVideoContainerWithView:_localVideoContainer];
    }else{
        // Show an alert, useless code
        UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Sorry" message:@"Unable to access camera"preferredStyle:UIAlertControllerStyleAlert];
        UIAlertAction *actionOk = [UIAlertAction actionWithTitle:@"Ok" style:UIAlertActionStyleDefault handler:nil];
        [alertController addAction:actionOk];
        [self presentViewController:alertController animated:YES completion:nil];
    }
}
- (void)rtcClientDidAddRemoteMediaStreamWithPeer:(RTCPeer *)peer stream:(RTCMediaStream *)stream audioOnly:(BOOL)audioOnly{
    [peer setRemoteVideoContainerWithView:_remoteVideoContainer];
    
}



@end

File: WebRTC-iOS-Client/WebRTC-iOS-Client/ViewController.swift
//
//  ViewController.swift
//  WebRTC-iOS-Client
//
//  Created by Innovation on 3/30/17.
//  Copyright © 2017 Disi A. All rights reserved.
//

import UIKit
import RTCSignaling

let serverUrl = "wss://iconthin.com:8443/ws/"
// let roomId = "abc"

class ViewController: UIViewController {
    
    var localVideoTrack: RTCVideoTrack? //Not yet used
    var remoteVideoTrack: RTCVideoTrack? //Not yet used
    var useSpeaker = false
    var videoSwitch = true
    var audioSwitch = true
    
    @IBOutlet weak var remoteVideoContainer: UIView!
    @IBOutlet weak var localVideoContainer: UIView!
    
    @IBOutlet weak var roomIdField: UITextField!
    
    @IBAction func connectClick(_ sender: Any) {
        roomIdField.resignFirstResponder()
        if let text = roomIdField.text {
            let roomId = text.replacingOccurrences(of: " ", with: "")
            if !roomId.isEmpty{
                RTCClient.shared.connect(serverUrl: serverUrl, roomId: roomId, delegate: self)
                return
            }
        }
        let alert = UIAlertController(title: "Sorry", message: "Please enter a valid roomId", preferredStyle: UIAlertControllerStyle.alert)
        alert.addAction(UIAlertAction(title: "OK", style: UIAlertActionStyle.default, handler: nil))
        self.present(alert, animated: true, completion: nil)
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }

    @IBAction func videoSwitchClick(_ sender: Any) {
        videoSwitch = !videoSwitch
        RTCClient.shared.setVideo(on: videoSwitch)
    }
    
    @IBAction func audioSwitchClick(_ sender: Any) {
        audioSwitch = !audioSwitch
        RTCClient.shared.setAudio(on: audioSwitch)
    }

    @IBAction func toggleSpeaker(_ sender: Any) {
        useSpeaker = !useSpeaker
        RTCClient.shared.setAudioOutput(useSpeaker: useSpeaker)
    }

}
extension ViewController: RTCClientDelegate{
    
    func rtcClientDidSetLocalMediaStream(client: RTCSignaling.RTCClient, authorized: Bool, audioOnly: Bool){
        if authorized{
            client.setLocalVideoContainer(view: localVideoContainer)
        }else{
            assertionFailure("Unauthorized to access media device")
        }
    }
    
    func rtcClientDidAddRemoteMediaStream(client: RTCSignaling.RTCClient, peer: RTCSignaling.RTCPeer, stream: RTCMediaStream, audioOnly: Bool){
        peer.setRemoteVideoContainer(view: remoteVideoContainer)
    }
    
    func rtcClientDidRemoveRemoteMediaStream(client: RTCSignaling.RTCClient, peer: RTCSignaling.RTCPeer, stream: RTCMediaStream, audioOnly: Bool){
        print("Remote peer did finish.")
    }

}

